nixpkgs_review/review.py

```diff
@@ -123,7 +123,7 @@ def worktree_dir(self) -> str:
         return str(self.builddir.worktree_dir)
 
     def git_merge(self, commit: str) -> None:
-        sh(["git", "merge", "--no-commit", commit], cwd=self.worktree_dir())
+        sh(["git", "merge", "--no-commit", "--no-ff", commit], cwd=self.worktree_dir())
```

Comment:
`--no-ff` or `--ff-only`?

Reply:
we want to set it to `--no-ff` because `--ff-only` might not work in some cases.

Reply:
Ok. I made the tests consistent with the actual code.

Reply:
Maybe to clarify more this is how I understand these options work in git.

- by default git automatically chooses between fast forward (`--ff`) or merge commit.
- when `--ff-only` git will fail if fast-forward is not possible (this is the setting I use because I like to know that it isn't if it isn't)
- when `--no-ff` git will not attempt to do fast-forward and just creates a merge commit even when fast-forward would be possible. This is likely the fastest and least likely to fail but I never really tried to measure it. If it's faster it will be hardly noticeable anyway.

I think the options are `--ff` and `--no-ff`. I chose the `--no-ff` as it seems more predictable to me.

[documentation](https://www.git-scm.com/docs/merge-options#Documentation/merge-options.txt---ff)

Reply:
> Ok. I made the tests consistent with the actual code.

thanks and sorry for that. I've completely missed the inconsistency there. :(